bot/exts/evergreen/wikipedia.py

```diff
@@ -1,114 +1,99 @@
-import asyncio
-import datetime
 import logging
+import re
+from datetime import datetime
+from enum import Enum
+from html import unescape
 from typing import List, Optional
 
-from aiohttp import client_exceptions
-from discord import Color, Embed, Message
+from discord import Color, Embed
 from discord.ext import commands
 
-from bot.constants import Wikipedia
+from bot.bot import Bot
+from bot.utils import LinePaginator
 
 log = logging.getLogger(__name__)
 
-SEARCH_API = "https://en.wikipedia.org/w/api.php?action=query&list=search&srsearch={search_term}&format=json"
-WIKIPEDIA_URL = "https://en.wikipedia.org/wiki/{title}"
+SEARCH_API = "https://en.wikipedia.org/w/api.php?action=query&list=search&prop=info&inprop=url&utf8=& \
+      format=json&origin=*&srlimit={number_of_results}&srsearch={string}"
+WIKI_THUMBNAIL = "https://upload.wikimedia.org/wikipedia/en/thumb/8/80/Wikipedia-logo-v2.svg" \
+                 "/330px-Wikipedia-logo-v2.svg.png"
+WIKI_SNIPPET_REGEX = r'(<!--.*?-->|<[^>]*>)'
+WIKI_SEARCH_RESULT = (
+    "**[{name}]({url})**\n"
+    "{description}\n"
+)
 
 
-class WikipediaSearch(commands.Cog):
-    """Get info from wikipedia."""
+class WikipediaSearchErrors(Enum):
+    """Errors returned in wikipedia search function."""
 
-    def __init__(self, bot: commands.Bot):
-        self.bot = bot
-        self.http_session = bot.http_session
+    no_results = "Sorry, we could not find a wikipedia article using that search term."
+    api_issue = "Whoops, the Wikipedia API is having some issues right now. Try again later."
```

Comment:
Since both the error message are used only once, they can be directly written in the error handling snippet.